// dao/baseDAO.js
var ObjectId = require('mongodb').ObjectID;
var dbConn = require('dao/connexion');
var coleccionSelecc = "preguntas";

module.exports = {
	insertar: function(docEntidad,callback){
		console.log('en proyectoDAO, docEntidad:');
		console.log(docEntidad);
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			insert(docEntidad, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	borrar: function(idBorrar,callback){
		console.log('en borrar de proyectoDAO');
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			remove({_id:new ObjectId(idBorrar)}, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	actualizar: function(docEntidad,callback){
		console.log('en actualizar dao');
		
		dbConn.conectar(function(db){
			var id = docEntidad._id;
			delete docEntidad._id;
			db.collection(coleccionSelecc).
			update({"_id" : new ObjectId(id)},docEntidad, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	buscar: function(idDocEntidad,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc)
			.findOne({ "_id" : new ObjectId(idDocEntidad) }, function(err,resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});   
		});
	},
	buscarTodos: function(callback){
		console.log("en proyectoDAO.js....");
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			find({}).toArray(function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	}
}
