// dao/contactoDAO.js
var ObjectId = require('mongodb').ObjectID;
var dbConn = require('dao/connexion');
var coleccionSelecc = "contactos";

module.exports = {
	insertar: function(docEntidad,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			insert(docEntidad, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	borrar: function(idBorrar,callback){
		dbConn.conectar(function(db){
			
			db.collection(coleccionSelecc).
			remove({_id:new ObjectId(idBorrar)}, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	actualizar: function(docEntidad,callback){
		dbConn.conectar(function(db){
			var id = docEntidad._id;
			delete docEntidad._id;
			db.collection(coleccionSelecc).
			update({"_id" : new ObjectId(id)},docEntidad, function(err, resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});      
		});
	},
	buscar: function(idDocEntidad,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc)
			.findOne({ "_id" : new ObjectId(idDocEntidad) }, function(err,resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});   
		});
	},	
	buscarCodigo: function(idDocEntidad,callback){
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc)
			.findOne({ "codigo" : new ObjectId(idDocEntidad) }, function(err,resultado) {
    			if(err) throw err;
    			callback(resultado);
  			});   
		});
	},
	buscarTodos: function(callback){
		//console.log('coleccionSelecc:'+coleccionSelecc);
		dbConn.conectar(function(db){
			db.collection(coleccionSelecc).
			find({}).toArray(function(err, resultado) {
    			if(err) {console.log('error en clienteDAO.js');
    				throw err;}
    			else
    			{console.log('exito en contactoDAO.js');
    			 //console.log(resultado);
    				callback(resultado);}
  			});      
		});
		
	}
}
